software/front/US_BootLoader.X/PIC32MX564F128H.c

```diff
@@ -141,17 +142,17 @@ UINT mcuWriteFlashRecord(void * RecordData)
     void* ProgAddress;
     void* ProgAddressEnd;
     UINT32 WrData;
-    UINT Result;
-    lFR.Address = ((UINT32 *)RecordData)[0];
-    lFR.RecDataLen = ((UINT32 *)RecordData)[1];
-    lFR.Data = (UINT8*)&((UINT32 *)RecordData)[3];
```

Comment:
Here the important part is lFR.Data = (UINT8*)&((UINT32 *)RecordData)[2];
I've made it in the previous format as I find it easier to read.

Reply:
IMHO, indexing in square braces (addr[index]) is more readable than indexing with math (addr+index), despite that the effect is the same. It seems to me that square braces makes it more obvious that the indexing depends on data size than math.

The real problem here was that:
lFR.Data = (UINT8*)&((UINT32 *)RecordData)[3];
should have been:
lFR.Data = (UINT8*)&((UINT32 *)RecordData)**[2]**;

And I think this is the thing that broke the bootloader.

Is this editable? I really want to keep the square brace indexing instead of pointer math here.

Reply:
No, I'll have to upload a second patch for it.

Reply:
Done.

Reply:
Merged it. Look OK, but please verify that it works. I won't have time for this in the following day or two.